### PR TITLE
Switched BINARY_OP Benchmarks from GoogleBench to NVBench

### PR DESCRIPTION
**Pull request by lamarrr**
## Description
This merge request switches the Benchmarking solution for the BINARY_OP benchmarks from GoogleBench to NVBench

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/benchmarks/binaryop/binaryop.cpp [hunk 1]
```diff
@@ -33,23 +32,21 @@ enum class TreeType {
...
   auto const n_cols       = reuse_columns ? 1 : tree_levels + 1;
   auto const source_table = create_sequence_table(
     cycle_dtypes({cudf::type_to_id<key_type>()}, n_cols), row_count{table_size});
   cudf::table_view table{*source_table};
 
-  // Execute benchmark
-  for (auto _ : state) {
-    cuda_event_timer raii(state, true);  // flush_l2_cache = true, stream = 0
+  // Use the number of bytes read from global memory
+  state.add_global_memory_reads<key_type>(table_size * (tree_levels + 1));
```
Comment (ttnghia) [inline cpp/benchmarks/binaryop/binaryop.cpp:47]:
```suggestion
  state.add_global_memory_reads<key_type>(table_size * (tree_levels + 1) * sizeof(key_type));
```

File: cpp/benchmarks/binaryop/compiled_binaryop.cpp [hunk 2]
```diff
@@ -38,31 +36,26 @@ void BM_compiled_binaryop(benchmark::State& state, cudf::binary_operator binop)
...
-    cuda_event_timer timer(state, true);
-    cudf::binary_operation(lhs, rhs, binop, output_dtype);
-  }
-
   // use number of bytes read and written to global memory
-  state.SetBytesProcessed(static_cast<int64_t>(state.iterations()) * column_size *
-                          (sizeof(TypeLhs) + sizeof(TypeRhs) + sizeof(TypeOut)));
+  state.add_global_memory_reads<TypeLhs>(table_size);
+  state.add_global_memory_reads<TypeRhs>(table_size);
+  state.add_global_memory_reads<TypeOut>(table_size);
```
Comment (ttnghia) [inline cpp/benchmarks/binaryop/compiled_binaryop.cpp:42]: `table_size` is number of rows, not size bytes. Need to multiply it with `sizeof(...)`.

**Review by ttnghia — APPROVED** [review]

**Review by kingcrimsontianyu — APPROVED** [review]
Lgtm. Thanks!

**Pull request merged**